.templates/setup.py.template

```diff
@@ -12,10 +13,21 @@ if "bdist_wheel" in sys.argv:
 else:
     # check if one of the tensorflow packages is already installed (so that we
     # don't force tensorflow to be installed if e.g. tensorflow-gpu is already
-    # there)
-    tf_dists = ["tf-nightly-gpu", "tf-nightly", "tensorflow-gpu"]
-    installed_dists = [d.project_name for d in pkg_resources.working_set]
-    for d in tf_dists:
+    # there).
+    # as of pep517 and pip>=10.0, pip will be running this file inside an isolated
+    # environment, so we can't just look up the tensorflow version in the current
+    # environment. but the pip package will be in the isolated sys.path, so we can use
+    # that to look up the site-packages directory of the original environment.
+    target_path = os.path.join("site-packages", "pip")
+    for path in sys.path:
+        if target_path in path:
+            source_path = [path[: path.index("pip")]]
+            break
+    else:
+        # fallback if we're not in an isolated environment (i.e. pip<10.0)
```

Comment:
We also go here if we're not using pep517, which would be the case if we didn't have the pyproject.toml file, right? Not saying we need to add a comment since we'll always have that file now, just making sure I understand it.

Reply:
Yep that's right.